### ISSUE TITLE
Enhancement request: Using WebAssembly should include guide on consuming multi-value

### ISSUE THREAD

**Issue by jzabinski-dolios** [tinygo-org/tinygo-site#232]
Go [supports](https://go.dev/tour/basics/6) the use of multiple results. [So does WebAssembly](https://webassembly.org/roadmap/). However, due to compiler limitations, TinyGo cannot yet produce WebAssembly that uses the supported WebAssembly syntax.

TinyGo uses a workaround of placing multiple values into linear memory for consumption by the embedder. The embedder must supply the address (a byte offset) in linear memory where the values are to be stored.

[This issue](https://github.com/tinygo-org/tinygo/issues/2512) contains a full explanation of how multi-value works in TinyGo, and an example of how to implement Javascript that consumes multiple values. The guide on Using WebAssembly should have a section describing this functionality.

**Comment by aykevl** [comment]
> However, due to compiler limitations, TinyGo cannot yet produce WebAssembly that uses the supported WebAssembly syntax.

No. These are not compiler limitations. We do this for backwards compatibility. I _think_ you can enable multi-value using something like `-llvm-features=multivalue` but I haven't tested this.

In any case, if this is documented we should probably put it in https://tinygo.org/docs/concepts/compiler-internals/calling-convention/.
